pkg/healthcheck/healthcheck_output.go

```diff
@@ -48,20 +52,142 @@ func (cr CheckResults) RunChecks(observer CheckObserver) bool {
+		fmt.Fprintln(wout, headerTxt)
+		fmt.Fprintln(wout, strings.Repeat("=", len(headerTxt)))
+		fmt.Fprintln(wout)
+	}
+}
+
+// RunExtensionsChecks runs checks for each extension name passed into the `extensions` parameter
+// and handles formatting the output for each extension's check. This function also handles
+// finding the extension in the user's path and runs it.
+func RunExtensionsChecks(wout io.Writer, werr io.Writer, extensions []string, flags []string, output string) bool {
+	if output != JSONOutput {
+		headerTxt := "Linkerd extensions checks"
+		fmt.Fprintln(wout)
+		fmt.Fprintln(wout, headerTxt)
+		fmt.Fprintln(wout, strings.Repeat("=", len(headerTxt)))
+	}
+
+	spin := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
+	spin.Writer = wout
+
+	success := true
+	for i, extension := range extensions {
+		var path string
+		args := append([]string{"check"}, flags...)
+		var err error
+		results := CheckResults{
+			Results: []CheckResult{},
+		}
+		extensionCmd := fmt.Sprintf("linkerd-%s", extension)
+
+		switch extension {
+		case "jaeger":
+			path = os.Args[0]
+			args = append([]string{"jaeger"}, args...)
+		case "viz":
+			path = os.Args[0]
+			args = append([]string{"viz"}, args...)
+		case "multicluster":
+			path = os.Args[0]
+			args = append([]string{"multicluster"}, args...)
+		default:
+			path, err = exec.LookPath(extensionCmd)
+			results.Results = []CheckResult{
+				{
+					Category:    CategoryID(extensionCmd),
+					Description: fmt.Sprintf("Linkerd extension command %s exists", extensionCmd),
+					Err:         err,
+					HintURL:     DefaultHintBaseURL + "extensions",
+					Warning:     true,
+				},
+			}
+		}
+
+		if err == nil {
+			if isatty.IsTerminal(os.Stdout.Fd()) {
+				spin.Suffix = fmt.Sprintf(" Running %s extension check", extension)
+				spin.Color("bold") // this calls spin.Restart()
+			}
+			plugin := exec.Command(path, args...)
+			var stdout, stderr bytes.Buffer
+			plugin.Stdout = &stdout
+			plugin.Stderr = &stderr
+			plugin.Run()
+			extensionResults, err := parseJSONCheckOutput(stdout.Bytes())
+			spin.Stop()
+			if err != nil {
+				command := fmt.Sprintf("%s %s", path, strings.Join(args, " "))
+				if len(stderr.String()) > 0 {
+					err = errors.New(stderr.String())
+				} else {
+					err = fmt.Errorf("invalid extension check output from \"%s\" (JSON object expected):\n%s\n[%s]", command, stdout.String(), err)
+				}
+				results.Results = append(results.Results, CheckResult{
+					Category:    CategoryID(extensionCmd),
+					Description: fmt.Sprintf("Running: %s", command),
+					Err:         err,
+					HintURL:     DefaultHintBaseURL + "extensions",
+				})
+				success = false
+			} else {
+				results.Results = append(results.Results, extensionResults.Results...)
+			}
+		}
+
+		var containsErr bool
+		for _, res := range results.Results {
+			if res.Err != nil {
+				containsErr = true
+				break
+			}
+		}
+
+		if output != JSONOutput && i < len(extensions) {
```

Comment:
Since `i` is the index of the range over extensions, won't `i < len(extensions)` always be true?  What is the intent of this check?

Reply:
Ah you are right, this isn't necessary. Good catch